packages/react-native/ReactAndroid/cmake-utils/ReactNative-application.cmake

```diff
@@ -81,39 +83,36 @@ target_link_libraries(${CMAKE_PROJECT_NAME}
         reactnative                         # prefab ready
 )
 
-# We use an interface target to propagate flags to all the generated targets
-# such as the folly flags or others.
+# Use an interface target to propagate flags to generated targets
 add_library(common_flags INTERFACE)
 target_compile_options(common_flags INTERFACE ${folly_FLAGS})
 
-# If project is on RN CLI v9, then we can use the following lines to link against the autolinked 3rd party libraries.
-if(EXISTS ${PROJECT_BUILD_DIR}/generated/autolinking/src/main/jni/Android-autolinking.cmake)
-        include(${PROJECT_BUILD_DIR}/generated/autolinking/src/main/jni/Android-autolinking.cmake)
-        target_link_libraries(${CMAKE_PROJECT_NAME} ${AUTOLINKED_LIBRARIES})
-        foreach(autolinked_library ${AUTOLINKED_LIBRARIES})
-            target_link_libraries(${autolinked_library} common_flags)
-        endforeach()
+# Autolinked libraries if available
+if(EXISTS ${BUILD_DIR}/generated/autolinking/src/main/jni/Android-autolinking.cmake)
+    include(${BUILD_DIR}/generated/autolinking/src/main/jni/Android-autolinking.cmake)
```

Comment:
Why are we moving from `PROJECT_BUILD_DIR` to just `BUILD_DIR`? Which issue does this solve?